conda_forge_tick/update_upstream_versions.py

```diff
@@ -88,28 +131,39 @@ def get_latest_version(meta_yaml):
             return ver
         else:
             with open('upstream_bad', 'a') as f:
-                f.write('{}: Could not find version on {} at {}\n'.format(
-                    meta_yaml['name'], source.name, url))
+                f.write('{}: Could not find version on {}\n'.format(
+                    meta_yaml['name'], source.name))
     with open('upstream_bad', 'a') as f:
-        f.write('{}: unknown source at {}\n'.format(meta_yaml['name'],
-                                                    meta_yaml['url']))
+        f.write('{}: unknown source\n'.format(meta_yaml['name']))
     return False
 
 
-# gx = nx.read_yaml('graph.yml')
-gx = nx.read_gpickle('graph.pkl')
+def update_upstream_versions(gx, sources=[PyPI(), CRAN(), Github()]):
```

Comment:
Should `RawURL` be in the sources? Also can the sources be a tuple (so we can avoid unintended mutation)?